poolmgr/gp.go

```diff
@@ -297,9 +316,23 @@ func (gp *GenericPool) specializePod(pod *v1.Pod, metadata *api.ObjectMeta) erro
 	// tell fetcher to get the function.
 	fetcherUrl := gp.getFetcherUrl(podIP)
 	log.Printf("[%v] calling fetcher to copy function", metadata.Name)
-	err := fetcherClient.DoFetchRequest(fetcherUrl, &fetcher.FetchRequest{
+
+	fn, err := gp.fissionClient.
+		Functions(metadata.Namespace).Get(metadata.Name)
+	if err != nil {
+		return err
+	}
+
+	pkg, err := gp.fissionClient.
```

Comment:
Why do we need this? You just have to pass the package metadata, you don't need the full package object. In fact you just need the Name and Namespace in the metadata.  (You could even change the fetchRequest to take a PackageRef instead of api.ObjectMeta, if you want to.)